content/en/docs/troubleshooting/DriftAndDiffDetection.md

```diff
@@ -0,0 +1,113 @@
+by a non-KubeFleet agent, KubeFleet will attempt to re-create it as soon as it finds out about the
+deletion.
+* Drift detection will not block resource rollouts. If you have just updated the resources on
+the hub cluster side and triggered a rollout, drifts on the member cluster side might have been 
+overwritten.
+* When a rollout is in progress, drifts will not be reported on the CRP status for a member cluster if
+the cluster has not received the latest round of updates.
+
+KubeFleet will check for configuration differences under the following two conditions:
+
+* When KubeFleet encounters a pre-existing resource, and the `whenToTakeOver` setting
+(`.spec.strategy.applyStrategy.whenToTakeOver` field) is set to `IfNoDiff`.
+* When the CRP uses an apply strategy of the `ReportDiff` type.
+
+Configuration difference details are reported in the CRP status
+on a per-cluster basis (`.status.placementStatuses[*].diffedPlacements` field). Note that the
+following limitations apply:
+
+* When you set the `comparisonOption` setting (`.spec.strategy.applyStrategy.comparisonOption` field)
+to `partialComparison`, KubeFleet will only check for configuration differences in managed fields,
+i.e., fields that have been explicitly specified on the hub cluster side. Unmanaged fields, such
+as additional labels and annotations, will not be considered as configuration differences.
+To check for such changes (field additions), use the `fullComparison` option for the `comparisonOption` field.
+* Depending on your cluster setup, there might exist Kubernetes webhooks/controllers (built-in or from a
+third party) that will process resources and add/modify fields as they see fit.
+The API server on the member cluster side might also add/modify fields (e.g., enforcing default values)
+on resources. If your comparison option allows, KubeFleet will report these as configuration differences.
+For any unexpected configuration difference reportings, verify first if you have installed a source that
+triggers the changes.
+* KubeFleet checks for configuration differences regardless of resource ownerships; resources not
+managed by KubeFleet will also be checked.
+* The absence of a resource will be considered as a configuration difference.
+* Configuration differences will not block resource rollouts. If you have just updated the resources on
+the hub cluster side and triggered a rollout, configuration difference check will be re-run based on the
+newer versions of resources.
+* When a rollout is in progress, configuration differences will not be reported on the CRP status
+for a member cluster if the cluster has not received the latest round of updates.
+
+Note also that drift detection and configuration difference check in KubeFleet run periodically.
+The reportings in the CRP status might not be up-to-date.
+
+## Investigation steps
+
+If you find an unexpected drift detection or configuration difference check result on a member cluster,
```

Comment:
how do users know that the detected drift is not expected? i.e. kubefleet didn't report a drift while someone has changed the resource in the member clusterr.

Reply:
Ah, I wrote this TSG mainly based on some of the Qs I've received from the C team and the I team before (when we are previewing the feature); they do have some confusion on cases like why labels/annotations added by K8s/third-party are ignored by Fleet's detection mechanism and why deletions are not reported.

Ryan, would you prefer if I add some explanation (commonly asked scenarios) about this part?